sparkmagic/sparkmagic/utils/configuration.py

```diff
@@ -0,0 +1,150 @@
+# Distributed under the terms of the Modified BSD License.
+import copy
+import sys
+import base64
+from hdijupyterutils.constants import EVENTS_HANDLER_CLASS_NAME, LOGGING_CONFIG_CLASS_NAME
+from hdijupyterutils.utils import join_paths
+from hdijupyterutils.configuration import override as _override
+from hdijupyterutils.configuration import override_all as _override_all
+from hdijupyterutils.configuration import with_override
+
+from .constants import HOME_PATH, CONFIG_FILE, MAGICS_LOGGER_NAME
+from .utils import get_livy_kind
+from sparkmagic.livyclientlib.exceptions import BadUserConfigurationException
+
+
+d = {}
+path = join_paths(HOME_PATH, CONFIG_FILE)
+
+    
+def override(config, value):
+    global d, path
+    _override(d, path, config, value)
+
+
+def override_all(obj):
+    global d
+    _override_all(d, obj)
+
+
+# Configs
+       
+def get_session_properties(language):
+    properties = copy.deepcopy(session_configs())
+    properties[u"kind"] = get_livy_kind(language)
+    return properties
+
+@with_override(d, path)
```

Comment:
Same comment as above: is calling `with_override(d, path)` every time necessary?